### PR TITLE
Remove BVHTree

### PR DESCRIPTION
**Pull request by kanye-quest**
# Summary

- Removes `spin::BVHTree`, now that `SignedDistance` uses `spin::BVH`
- Initial documentation of `spin::BVH`
- Bugfix for `spin::BVH::writeVtkFile()`

### CODE REVIEW

**Review by white238 — APPROVED** [review]
Looks great! thanks for continuing to lower our technical debt!

File: src/axom/spin/docs/sphinx/bvh.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+the child node, and all of its corresponding elements, can be skipped during the
+BVH traversal.  If the probe does intersect a child node, the node's children are also
+tested for probe intersection.  Without the acceleration data structure, each
+probe point must be tested against each triangle.
+
+.. literalinclude:: ../../examples/spin_introduction.cpp
+   :start-after: _bvh_candidate_start
+   :end-before: _bvh_candidate_end
+   :language: C++
+
```
Comment (agcapps) [inline src/axom/spin/docs/sphinx/bvh.rst:41]: Please add a Note or Warning here to remind users to deallocate the `candidates` array that `BVH::findPoints` (and related methods) allocates.  I see it's in the Doxygen, I see it's also in a comment in the `_bvh_candidate` section you just excerpted, but it's worth repeating.

**Review by agcapps — APPROVED** [review]
Looks very good.  I would like you to make one amplification in the docs, but besides that it's great.

**Review by rhornung67 — APPROVED** [review]
Nice work! 👍

**Review by kennyweiss — APPROVED** [review]
Thanks @kanye-quest !

**Pull request merged**